src/Features/Core/Portable/FeaturesResources.resx

```diff
@@ -1352,4 +1352,16 @@ This version used in: {2}</value>
   <data name="Make_field_readonly" xml:space="preserve">
     <value>Make field readonly</value>
   </data>
+  <data name="Assignment_can_be_simplified" xml:space="preserve">
+    <value>Assignment can be simplified</value>
+  </data>
+  <data name="Simplify_assignment" xml:space="preserve">
+    <value>Simplify assignment</value>
+  </data>
+  <data name="Return statement can be simplified" xml:space="preserve">
+    <value>Return statement can be simplified</value>
+  </data>
+  <data name="Simplify_return_statement" xml:space="preserve">
+    <value>Simplify return statement</value>
```

Comment:
I'm not sure if it's worth to have 2 separate messages for this. I woud imagine something like either "Simplify if statement" or "Use conditional expression". But maybe that's just because I'm thinking of the ways this could be generalized, and having a special message for other cases like "Simplify expression" would not make sense at that point.

Reply:
I would imagine settings similar to what you did in your parentheses work: "Prefer conditional expression in assignment", "in return statement" and "in other operations". But I don't see why it's needed to have different messages for these.

Reply:
An example of "other" might be:
```c#
if (condition)
    Console.WriteLine(1 + 2);
else
    Console.WriteLine(1 + 3);
```
to
```c#
Console.WriteLine(1 + (condition ? 2 : 3));
```
I can't come up with a special message for all these cases :/ "Simplify add expression"? In my view these should all have 1 message mentioning either the if statement or conditional expression.

Reply:
Sure, i'll look into seeing if this can be more generalized.

Reply:
Could you start an issue about this other feature?  I would prefer that this PR not be cluttered up by the speculative discussion of what we might do there.

Reply:
Also, I think it would be good to have a message mentioning "conditional expression" in these regardless of whether we talk about generalizing. "Simplify assignment" doesn't really tell you what the feature is about. If I saw a feature called "simplify assignment", I would probably think it did this:
```c#
a = a + 2;
```
to
```c#
a += 2;
```

Reply:
As I said, I think either "Simplify if statement" or just "Use conditional expression" would be a better name. With the latter, you'd know exactly what the feature is about just by seeing the message.

Reply:
"Simplify X" is usually not very descriptive. Especially for a code fix - it might be more acceptable for an analyzer.

Reply:
I tried to generalize this a bit.

Reply:
We use 'Simplify' in several other cases.

Reply:
@CyrusNajmabadi That's why it would be easy to get in a naming conflict with other features. "Simplify assignment" could apply to anything that somehow manipulates assignment expressions. If we take that name for this, what would you call a feature changing `a = a + 2` to `a += 2`? But I get it that you could still say this is just speculative discussion of what we might do in the future.

Reply:
Yes.  This is all speculative.

Reply:
In general we want to strike a reasonable balance between conveying what type of thing we're going to be doing, without having to be overly verbose and unwieldy with what we say.  We allow the preview "to do the talking" as it were.

So, we've tended to opt for simple high-level verbs to convey ideas.  "Convert to 'a'",  "Simplify 'b'" "Use 'c'" "Generate 'd'", "Make 'e'" etc. etc.

We actually had a bunch of messages that used to try to be more descriptive.  They actually just ended up being way to verbose and unwieldy and cluttered up the UI even *more* than what we ahve today.  In particular we wanted people to know things like "what are we making *and* where is it going?"  So we had things like "Generate field 'Dictionary<int, string> mappings' in class 'MyNamespace.Whatever'".  We pulled way back from that and just settled on letting users know the general idea of what we were going to do, but obviously letting them have hte preview so they could deeply deconstruct what was actually happening.

Reply:
I can understand that, but I still think "Use conditional expression" is vastly more descriptive without being more verbose.

Reply:
In my opinion, "Simplify assignment" isn't even on the same level of descriptiveness as "Generate class" - here you know what kind of thing is going to happen (even though it's without more details), but there are likely more ways how assignment could be simplified and I simply don't know what it would mean.

Reply:
On the other hand, I think it's perfectly fine to have a feature called "Simplify default expression", where it's pretty clear what this will do as there aren't a lot of other options / other forms of a default expression.